src/tokenomics/get-tokenomics-history.ts

```diff
@@ -0,0 +1,246 @@
+
+  // register types before creating the api
+  registerJoystreamTypes();
+
+  // Create the API and wait until ready
+  const api = await ApiPromise.create({provider});
+  let oldHash: Hash = await api.rpc.chain.getBlockHash(firstBlock-1)
+  const firstBlockHash: Hash = await api.rpc.chain.getBlockHash(firstBlock)
+
+  let councilElectedAtFirstblock = false
+  let councilTermEndedBlock = lastBlock
+
+  const valRewards: ValidatorReward[] = []
+  const slashings: Slashing[] = []
+  let lastValRewardHeight = await getLastStakingRewardHeight(api, firstBlock)
+  const proposalIds: number[] = []
+  let transactionFees = 0
+  let storageFees = 0
+  let membershipFees = 0
+  let balanceTransfers = 0
+  let extrinsicTipsPayed = 0
+  let actualExchangeBurns = 0
+
+  
+  //Get active proposals at start
+  const activeProposals = await getActiveProposals(api,firstBlockHash)
+
+  //Get fiat pool info at start
+  const poolStart = await getPoolStart(api, burnAddress, firstBlock, topUps, allExchanges)
+  const newExchanges: ExchangeInfo[] = []
+
+  //Iterate through all blocks to get all events (some things are only available this way)
+  for (let blockHeight=firstBlock; blockHeight<lastBlock+1; blockHeight++) {
+    const blockHash: Hash = await api.rpc.chain.getBlockHash(blockHeight)
+    const session = await api.query.session.currentIndex.at(blockHash) as SessionIndex
+    const era = await api.query.staking.currentEra.at(blockHash) as EraIndex
+    const events = await api.query.system.events.at(blockHash) as Vec<EventRecord>;
+    const getBlock = await api.rpc.chain.getBlock(blockHash) as SignedBlock
+    // Get extrinsic info
+    const extrinsicsInBlock = getBlock.block.extrinsics as Vec<Extrinsic>
+    for (let i=0; i<extrinsicsInBlock.length; i++) {
+      const sectionName = extrinsicsInBlock[i].method.sectionName
+      const methodName = extrinsicsInBlock[i].method.methodName
+      // Get tokens burned as "tips"
+      if (extrinsicsInBlock[i].tip.toNumber()>0) {
+        if (extrinsicsInBlock[i].signer.toString() === burnAddress) {
+          actualExchangeBurns += extrinsicsInBlock[i].tip.toNumber()
+        } else {
+          extrinsicTipsPayed += extrinsicsInBlock[i].tip.toNumber()
+        }
+      }
+      // Get tokens burned from tx fees
+      if (
+        sectionName === "actors" || sectionName ==="contentWorkingGroup" || sectionName === "councilElection" || 
+        sectionName === "dataDirectory" || sectionName === "dataObjectStorageRegistry" || sectionName === "discovery" || 
+        sectionName === "forum" || sectionName === "memo" || sectionName === "proposalsCodex" || 
+        sectionName === "proposalsDiscussion" || sectionName === "proposalsEngine" || sectionName === "staking" || 
+        sectionName === "versionedStorePermissions" ||
+        (sectionName === "members" && methodName !== "buyMembership") ||
+        (sectionName === "session" && methodName=== "setKeys")
+      ) {
+        transactionFees ++
```

Comment:
assumption that transaction fee is always 1 token is okay, but we will miss this when that changes.

in network integration tests there are some examples on how tx fees are calculated. It could get more complicated in future if weights of transactions also affect the transaction fee (like gas).

To keep it very simple I would have some function at the top `function getFeePerTransaction() {  return 1 }`,

and instead of `transactionFees++` do `transactionFees += getFeePerTransaction()` so it will be easy to update later